### ISSUE TITLE
Notification Handler Fails when Unable To Read Server Settings

### ISSUE THREAD

**Issue by johnfromthefuture** [simcen/alert_manager#140]
Attempting to update incidents as a non-admin user failed. After digging into it, an issue was related to an unhandled exception. In short: Saving an incident update invokes the incident_workflow.py script. This invokes the EventHandler; the EventHandler invokes the NotificationHandler.

I modified this code snippet in the Notification Handler:
```
        self.settings = {    
                            "MAIL_SERVER": server_settings['mailserver'],
                            "EMAIL_HOST_USER": server_settings['auth_username'],
                            "EMAIL_HOST_PASSWORD": clear_pass,
                            "EMAIL_USE_TLS": use_tls,
                            "EMAIL_USE_SSL": use_ssl
                        }

```
to this:
```
mailserver = ''
        if 'mailserver' in server_settings:
            mailserver = server_settings['mailserver']

        auth_username = ''
        if 'auth_username' in server_settings:
            auth_username = server_settings['auth_username']

        self.settings = {    
                            "MAIL_SERVER": mailserver,
                            "EMAIL_HOST_USER": auth_username,
                            "EMAIL_HOST_PASSWORD": clear_pass,
                            "EMAIL_USE_TLS": use_tls,
                            "EMAIL_USE_SSL": use_ssl
                        }
```

Without that change, the users required more permissions than I wanted to give.

**Comment by jdeer0618** [comment]
Had same issue. Used @johnfromthefuture fix and my "user" role can edit incidents.

**Comment by simcen** [comment]
What version are you guys using?
In the current develop branch, the relevant code part looks like
`        # Configure django settings
        clear_pass = ''
        if 'clear_password' in server_settings:
            clear_pass = server_settings['clear_password']

        auth_username = ""
        if 'auth_username' in server_settings:
            auth_username = server_settings['auth_username']

        mail_server = "localhost"
        if 'mailserver' in server_settings:
            mail_server = server_settings['mailserver']

        self.settings = {    
                            "MAIL_SERVER": mail_server,
                            "EMAIL_HOST_USER": auth_username,
                            "EMAIL_HOST_PASSWORD": clear_pass,
                            "EMAIL_USE_TLS": use_tls,
                            "EMAIL_USE_SSL": use_ssl
                        }`

@johnfromthefuture Could you describe what exactly was your fix? I didn't really understand your snippet, sorry :)

**Comment by johnfromthefuture** [comment]
The snippet you posted from the development branch would fix the issue and is more-or-less the same thing I did. :) As to the version that I'm on... it is 2.0.4 but heavily modified. I try to backport fixes released in your development and bring up bugs that I notice.

The short story there is that I'm SOC building and using Alert Manager as the core incident tracking mechanism. I ran into two major problems though: a lack of exception handling at some key points caused incidents to be lost; issues with the KV store for high throughput/lots of incidents.

I'll see about checking out the latest development version after .conf next week and contributing that way. It's the least I can do.
